### PR TITLE
Fixes #8353  - Implement Aggregation Endpoint for dataInsight

### LINKED ISSUE #8353: Implement Data Insight aggregation endpoint
### Context
In order to consume fromatted data to display in graphs we need to implement an aggregation endpoint that will perform computation against ES indexes

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :

Fixes #8353

**Note:** We only have the entity data insight chart aggregation in this PR. We'll be adding the overview and webanalytics in a seperate PR.

This PR implements the data insight endpoint. The aggregation logic is tight to a data insight chart type. A data insight chart type holds metadata information about a chart (e.g. description, name, display name, dimensions, metrics, etc.).

To get aggregated data for a chart we'll need to pass at least:
- startTs (milliseconds)
- endTs (milliseconds)
- dataInsightChartName (the chart for which we are aggreagting data)
- dataReportIndex (the index in which we'll be fetching the data) -> this information can be retrieve from the dataChartInsight entity
- team (optional to filter for specific team)
- tier (optional to filter for specific tier)

Sample request
```
curl --location --request GET 'http://localhost:8585/api/v1/dataInsight/aggregate?startTs=1667154603827&endTs=1667239204186&dataInsightChartName=TotalEntitiesByType&dataReportIndex=entity_report_data_index' --header 'Authorization: Bearer <Token>'
```

Additional Information
To add sample data you'll need to start the server and add some entities (you can also start the server with `sh docker/run_local_docker.sh`). You can add tiers and teams to the entities. Then run the following python integration tests 

```
python -m pytest /<path>/<to>/OpenMetadata/ingestion/tests/integration/data_insight/test_data_insight_workflow.py::DataInsightWorkflowTests::test_execute_method
```

Note that we are aggregating data daily, so running this command above in within the same-day interval will add data for the same day.

Additional Work shipped in this PR
- metadata_entity -> these are metadata_entities that is used as a service
- Elasticsearch info will be added to a sink param in the IngestionPipeline automatically if type is `dataInsight`

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

**Comment by TeddyCr** [comment]
cc @Sachin-chaurasiya

**Comment by TeddyCr** [comment]
cc @mohityadav766

File: openmetadata-service/src/main/java/org/openmetadata/service/Entity.java [hunk 1]
```diff
@@ -99,6 +99,8 @@ public final class Entity {
   public static final String TEST_SUITE = "testSuite";
   public static final String TEST_CASE = "testCase";
   public static final String WEB_ANALYTIC_EVENT = "webAnalyticEvent";
+  public static final String DATA_INSIGHT_CHART = "dataInsightChart";
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/Entity.java:102]: should we call this metric or chart?

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/metadata/MetadataResource.java [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+import org.openmetadata.service.jdbi3.ListFilter;
+import org.openmetadata.service.jdbi3.MetadataRepository;
+import org.openmetadata.service.resources.Collection;
+import org.openmetadata.service.resources.EntityResource;
+import org.openmetadata.service.security.Authorizer;
+import org.openmetadata.service.util.RestUtil;
+import org.openmetadata.service.util.ResultList;
+
+@Slf4j
+@Path("/v1/metadata")
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/metadata/MetadataResource.java:53]: can we push this under the same path like /v1/anayltics/metadata /v1/analytics/aggregate

Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/metadata/MetadataResource.java:53]: Sounds good. Should I still keep the dataInsight resource?

Also the metadata resource will be removed as @mohityadav766 will be handling this (from our discussion yesterday in the eng. meeting)

File: openmetadata-service/src/main/java/org/openmetadata/service/dataInsight/DataInsightAggregatorFactory.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+public class DataInsightAggregatorFactory {
+
+  public DataInsightAggregatorFactory() {}
+
+  public static DataInsightAggregatorInterface createDataAggregator(
+      Aggregations aggregations, DataInsightChartResult.DataInsightChartType dataInsightChartType)
+      throws IllegalArgumentException {
+    switch (dataInsightChartType) {
+      case PERCENTAGE_OF_ENTITIES_WITH_DESCRIPTION_BY_TYPE:
```
[the remark this answers is not in the document]
Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/dataInsight/DataInsightAggregatorFactory.java:14]: Makes sense. I'll need to think about how we can do that as we have some charts that are going to be quite different (i.e. web analytics vs entity aggregation).

From what I can see Common room uses graphql. An actual query is sent to fetch the data.

File: openmetadata-service/src/main/java/org/openmetadata/service/Entity.java (hunk 1 above)
Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/Entity.java:102]: I used chart it represents the actual chart. Inside a chart we have metrics and dimensions

File: openmetadata-service/src/main/resources/json/data/dataInsight/totalEntitiesByTier.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "TotalEntitiesByTier",
+  "displayName": "Percentage of Entities With Owner",
+  "description": "Display the percentage of entities with owner by type.",
```
Comment (Sachin-chaurasiya) [inline openmetadata-service/src/main/resources/json/data/dataInsight/totalEntitiesByTier.json:4]:
```suggestion
  "displayName": "Percentage of Entities With Tier",
  "description": "Display the percentage of entities with tier by type.",
```

File: openmetadata-service/src/main/resources/json/data/dataInsight/totalEntitiesByType.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "TotalEntitiesByType",
+  "displayName": "Percentage of Entities With Owner",
+  "description": "Display the percentage of entities with owner by type.",
```
Comment (Sachin-chaurasiya) [inline openmetadata-service/src/main/resources/json/data/dataInsight/totalEntitiesByType.json:4]:
```suggestion
  "displayName": "Total Entities",
  "description": "Display the total of entities by type.",
```

**Review by Sachin-chaurasiya — APPROVED** [review]
Hello @TeddyCr, some files have duplicate text, please update them.

File: openmetadata-service/src/main/java/org/openmetadata/service/dataInsight/DataInsightAggregatorFactory.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (harshach) [reply openmetadata-service/src/main/java/org/openmetadata/service/dataInsight/DataInsightAggregatorFactory.java:14]: Es can have graphql froented but probbaly overkill. 
ES also have sql query like syntax to run aggregations on various dimensions. We can keep web analytics separate for entity aggregation.
The way I see it is you have 2 new mappings one for web analytics and another es analytics.  Each of those mappings should get indexed across different dimensions that way we can run aggregates across different dimensions

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html

**Review by harshach — DISMISSED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**